Designer job posting for the reproducibility project (#227)
OpenRefine/openrefine.org: blog/2023-07-13-reproducibility-designer.md

---
author: Antonin Delpeuch

> title: "Designer seeked for OpenRefine's reproducibility improvements"

Reviewer: ```suggestion
title: "Designer sought for OpenRefine's reproducibility improvements"
```
although using active voice instead of passive would be even better e.g. We're hiring a workflow designer!

Author: Active voice makes sense, but I would not qualify the position as a "workflow designer", because that feels a bit too narrow (there are reproducibility issues in scope which are not strictly about the workflow language). Also I think it could scare off people who would be good matches for the role.

---

OpenRefine is hiring a designer to help shape our improvements to OpenRefine's operation history and test those changes with user panels.
<!--truncate-->

> [OpenRefine](https://openrefine.org/) is a powerful tool to clean messy data, popular in a diverse range of communities. It has been serving the needs of journalists, librarians, Wikipedians, scientists for more than ten years and is taught in many curricula and workshops around the world. OpenRefine is a fiscally sponsored project of [Code for Science & Society Inc](https://codeforscience.org/), a 501(c)(3) charitable organization in the USA.

Reviewer: ```suggestion
[OpenRefine](https://openrefine.org/) is a power tool for working with messy data, popular in a diverse range of communities. It has been serving the needs of journalists, librarians, Wikipedians, scientists for more than 13 years and is taught in many curricula and workshops around the world. OpenRefine is a fiscally sponsored project of [Code for Science & Society Inc](https://codeforscience.org/), a 501(c)(3) charitable organization in the USA.
```

Much of the power of a tag line in branding comes from repetition. If it's going to be changed, it should be changed consistently everywhere. Ten should be written as digits according to most style guides (but it's actually 13+).

Author: Thanks!

> We are currently working on improving the reproducibility of OpenRefine workflows as part of [a project funded by the Essential Open Source Software for Science programme](https://chanzuckerberg.com/eoss/proposals/improving-openrefines-reproducibility/).

Reviewer: OpenRefine doesn't have "workflows." It has an Undo history which was inadvisably advertised as a pseudo-workflow-like capability without any of the design or engineering to make it actually work as a workflow.

I think it would be better to recast the job description in this light (ie it's a new design task, not "fixing" an old design)

Author: I've changed the wording here, can you have a look if it suits you?

Reviewer: That looks better. I took a crack at editing it a little bit more.

> This consists in improving the user experience around OpenRefine's operation history, which tracks all transformations applied on the original data. This work aims to address a range of long standing issues, such as:

Reviewer: ```suggestion
This consists of improving the user experience around OpenRefine's operation history, which tracks all transformations applied on the original data. This work aims to address a range of long standing issues, such as:
```

- limitations of the Undo/Redo functionality, such as the lack of ability to undo an operation in the middle of the history list ([#183](https://github.com/OpenRefine/OpenRefine/issues/183)) or unintentional loss of work after using the undo feature and
  applying a new operation ([#369](https://github.com/OpenRefine/OpenRefine/issues/369), [#3184](https://github.com/OpenRefine/OpenRefine/issues/3184))
- the difficulty in reusing and sharing OpenRefine workflows, due to the lack of error handling when applying series of operations and the difficulty to adapt a workflow to a new project (for instance, with different column names)

> - the lack of support for running OpenRefine workflows as part of larger pipelines, without using the web UI interactively. This covers for instance the lack of import metadata in the JSON export of the history

Reviewer: ```suggestion
- the lack of support for running OpenRefine workflows as a part of larger pipelines, without using the web UI interactively. This covers, for instance, the lack of import metadata in the JSON export of the history
```

copy edit only, but this gets into the slippery slope of protocol versioning, command line clients, etc.

  ([#460](https://github.com/OpenRefine/OpenRefine/issues/460)).

See the corresponding [GitHub project](https://github.com/orgs/OpenRefine/projects/6) for more issues in the scope of this project.

> We are looking for a part-time designer who will

Reviewer: Part time covers a wide range. What is the expected commitment in hours per week/month?

Reviewer: I'm guessing folks want to leave this open ended, which is OK with me. It can be resolved during candidate discussions.

> * design changes to the user interface to support those improvements;

Reviewer: One of the first things which needs to be done, in my opinion, is to design (or, preferably, choose) a DSL which meets the (implied) requirements above of being amenable to source control diffing & merging, easy(ier) for a human to edit, compatible with a parser which can generate high quality error messages for the user, etc.

Author: That makes sense, I'd be happy to list that explicitly. In my mind it is indeed part of the role but perhaps not the first task I would give them because it's feels more challenging than UI tweaks, especially for someone new-ish to the project.

Author: I've added a bullet point about this. I have not used the term "DSL" because I think it could be seen as too programmer-oriented, which could scare off good candidates (we're not expecting the designer to design a programming language).
But I agree that from a technical perspective this is an appropriate term to describe what we are trying to build.

Reviewer: DSL wasn't intended for the copy. Since the domain is workflows, I'd probably have gone with "workflow language" rather than "representation of series of operations forming workflows." If you want to leave the door open for visual/graphical languages, perhaps just say that explicitly, "workflow language (textual or visual)."

I'll mark this as resolved and make a suggestion on the new verbiage.

* validate those choices by conducting user testing sessions;
* help prioritize follow-up improvements based on user feedback.

Requirements: 
* Experience with designing and testing software
* Ability to work in a fully remote environment
* Good communication skills and fluency in English

Nice to have:

> * Familiarity with OpenRefine

Reviewer: I'm struggling to see how this can be optional for a successful candidate. They'd at least need strong familiarity with one or more of other data wrangling software, workflow languages, or related topics.

Author: Yes, I also see it as a rather important skill, but my intention with classifying it as an optional skill is to broaden a bit the pool of potential applicants. I thought it's worth being open minded about the skills that someone would be bringing to the table: I can imagine someone with sufficient work experience and a solid methodology can be a good match for this role even if they are not familiar yet with the tool itself, as they'd be able to rely on the rest of the team for subtle product decisions but be more of a driving force around the overall design and testing process.

But overall I don't have strong feelings about how this should be worded / classified.

Reviewer 2: I kind of agree that familiarity with OpenRefine or some other data wrangling software should be required and not just a nice to have except the role makes room for a duration of onboarding because from experience, OpenRefine is not really the easiest of tools for a designer to get familiar with in a short time frame and especially on their own and this project seldom feels like a beginner's forte.

Author: I've moved this point to the requirements section, making it a bit lighter by accepting familiarity with other data cleaning tools as well.

Reviewer 3: Let's be honest - how many people in the existing OR community are designers? If this is a requirement, you will get very few candidates and those would be probably developers with interest in design, rather than designers per se. But maybe few candidates is OK, if you have limited time and scope for recruiting.

Reviewer: > I've moved this point to the requirements section, making it a bit lighter by accepting familiarity with other data cleaning tools as well.

Thanks. I'm happy with broadening further this to include other relevant experience such as workflow managers (Galaxy, Nextflow, Airflow, CWL) in addition to data wrangling tools.

* Experience with open source development practices and tools (such as the use of GitHub Issues)

**TODO compensation details**

> This role is available immediately and is initially expected to span the remaining duration of this project around reproducibility (1.5 year). We hope to be able to extend this role to work on other areas of the tool as other opportunities arise.

Reviewer: A 1 1/2 year initial term sounds like a lot. It's much easier to extend contracts than to cut them short. I'd recommend six month renewable (normally I might suggest 90 days, but I think there'll be a fair amount of learning curve and startup overhead).

Author: That makes sense. My motivation with advertising the full grant duration is to make it clear we are aiming to retain the person if everything works out nice. 6 months renewable sounds fine.
